fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaActivity.java

```diff
@@ -0,0 +1,210 @@
+                    .setTitle(String.format("popstellar - %s", newLaoName));
+              }
+            });
+
+    // Subscribe to "open home" event
+    mViewModel
+        .getOpenHomeEvent()
+        .observe(
+            this,
+            booleanEvent -> {
+              Boolean event = booleanEvent.getContentIfNotHandled();
+              if (event != null) {
+                setupSocialMediaHomeFragment();
+              }
+            });
+
+    // Subscribe to "open search" event
+    mViewModel
+        .getOpenSearchEvent()
+        .observe(
+            this,
+            booleanEvent -> {
+              Boolean event = booleanEvent.getContentIfNotHandled();
+              if (event != null) {
+                setupSocialMediaSearchFragment();
+              }
+            });
+
+    // Subscribe to "open following" event
+    mViewModel
+        .getOpenFollowingEvent()
+        .observe(
+            this,
+            booleanEvent -> {
+              Boolean event = booleanEvent.getContentIfNotHandled();
+              if (event != null) {
+                setupSocialMediaFollowingFragment();
+              }
+            });
+
+    // Subscribe to "open profile" event
+    mViewModel
+        .getOpenProfileEvent()
+        .observe(
+            this,
+            booleanEvent -> {
+              Boolean event = booleanEvent.getContentIfNotHandled();
+              if (event != null) {
+                setupSocialMediaProfileFragment();
+              }
+            });
+  }
+
+  public static SocialMediaViewModel obtainViewModel(FragmentActivity activity) {
+    return new ViewModelProvider(activity).get(SocialMediaViewModel.class);
+  }
+
+  @Override
+  public boolean onCreateOptionsMenu(Menu menu) {
+    getMenuInflater().inflate(R.menu.social_media_top_menu, menu);
+
+    // Get the submenu and clear its unique item. The item was needed to create the submenu
+    SubMenu laosList = menu.findItem(R.id.laos_list).getSubMenu();
+
+    // Adding all currently opened lao name to the submenu
+    mViewModel
+        .getLAOs()
+        .observe(
+            this,
+            list -> {
+              if (list != null) {
+                laosList.clear();
+                for (int i = 0; i < list.size(); ++i) {
+                  // Creating a unique id using laos_list and laos ids such that it doesn't override
+                  // them in onOptionsItemSelected
+                  laosList.add(Menu.NONE, R.id.laos_list + R.id.laos + i, i, list.get(i).getName());
+                }
+              }
+            });
+
+    return true;
+  }
+
+  @SuppressLint("NonConstantResourceId")
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    // Retrieve the index of the lao within the list
+    int i = item.getItemId() - R.id.laos_list - R.id.laos;
+    if (mViewModel.getLAOs().getValue() != null && i >= 0) {
+      Lao lao = mViewModel.getLAOs().getValue().get(i);
```

Comment:
i >= 0 was checked but not i < laos.size
```suggestion
    List<Lao> laos = mViewModel.getLAOs().getValue();
    if (laos != null && i >= 0 && i < laos.size()) {
      Lao lao = laos.get(i);
```